floor/floor/controller/midi/manager.py

```diff
@@ -88,11 +88,15 @@ def on_midi_commands(self, peer, commands):
         for command in commands:
             func = mapping.get_function(command)
             if not func:
+                self.logger.info("Received unmapped command: {} {}".format(command[0], command[1]))
                 continue
 
             value = command[2]
 
-            func.callback(self.controller, value)
+            try:
+                func.callback(self.controller, value)
+            except AttributeError as e:
```

Comment:
I noticed that pressing a mapped button that was not handled by the controller killed everything.  Given that I hope people will be mashing on this APCmini, I don't want random button presses to crash the floor.

Reply:
got a stack trace for this? feels like root issue is deeper in the stack.

Reply:
It was pretty straightforward.  Wrong function used in the mapping file.  Made an update here.

Reply:
Ah, in that case, let's catch `Exception`; after all, a future (mis-)implementation could trigger a different kind of exception and we'd need the same protection.

Also, `logger.exception` from within an exception scope will automatically include the stack trace in the log message.

```py
try:
  func.callback(self.controller, value)
except Exception:
  self.logger.exception('Error in MIDI command callback')
```

Reply:
Yah, I started with `Exception` and then PyCharm complained that it was too general, and then I googled to see if py-people still felt that a general catch was evil and Python is pure and each exception type is precious and should be handled like a gift, and found it still to be true and I yelled at the computer screen about it and then changed it to a more specific type anyway.

Maybe I'll change it back.